src/mscorlib/shared/Interop/Unix/System.Native/Interop.PathConf.cs

```diff
@@ -24,50 +24,7 @@ internal enum PathConfName : int
             PC_VDISABLE         = 9,
         }
 
-        /// <summary>The maximum path length for the system.  -1 if it hasn't yet been initialized.</summary>
-        private static int s_maxPath = -1;
-
-        /// <summary>The maximum name length for the system.  -1 if it hasn't yet been initialized.</summary>
-        private static int s_maxName = -1;
-
-        internal static int MaxPath
-        {
-            get
-            {
-                // Benign race condition on cached value
-                if (s_maxPath < 0) 
-                {
-                    // GetMaximumPath returns a long from PathConf
-                    // but our callers expect an int so we need to convert.
-                    long temp = GetMaximumPath();
-                    if (temp > int.MaxValue)
-                        s_maxPath = int.MaxValue;
-                    else
-                        s_maxPath = Convert.ToInt32(temp);
-                }
-                return s_maxPath; 
-            }
-        }
-
-        internal static int MaxName
```

Comment:
Is this one worth removing? I don't want to hunt down finding max filename issues in the future if we find a filesystem being supported that has more than 255 character filenames (unlikely as that is).

Reply:
Its usage as a global max is bad. `PathConf(path, PC_NAME_MAX)` should be used instead.
There shouldn't be much reasons to use this, the system library/os will do checks and return error codes.
With this PR and corresponding one in corefx, the only use of this is in WaitHandleNameMax as part of a message to inform the user.